include/triton/Tools/Sys/GetPlatform.hpp

```diff
@@ -0,0 +1,300 @@
+
+  // Get Num Shader Engines
+  err = hsa_agent_get_info(
+      agent, (hsa_agent_info_t)HSA_AMD_AGENT_INFO_NUM_SHADER_ENGINES,
+      &agent_i->shader_engs);
+
+  // Get Num Shader Arrays per Shader engine
+  err = hsa_agent_get_info(
+      agent, (hsa_agent_info_t)HSA_AMD_AGENT_INFO_NUM_SHADER_ARRAYS_PER_SE,
+      &agent_i->shader_arrs_per_sh_eng);
+
+  // Get number of Compute Unit
+  err = hsa_agent_get_info(
+      agent, (hsa_agent_info_t)HSA_AMD_AGENT_INFO_COMPUTE_UNIT_COUNT,
+      &agent_i->compute_unit);
+
+  // Check if the agent is kernel agent
+  if (agent_i->agent_feature & HSA_AGENT_FEATURE_KERNEL_DISPATCH) {
+    // Get flaf of fast_f16 operation
+    err = hsa_agent_get_info(agent, HSA_AGENT_INFO_FAST_F16_OPERATION,
+                             &agent_i->fast_f16);
+
+    // Get wavefront size
+    err = hsa_agent_get_info(agent, HSA_AGENT_INFO_WAVEFRONT_SIZE,
+                             &agent_i->wavefront_size);
+
+    // Get max total number of work-items in a workgroup
+    err = hsa_agent_get_info(agent, HSA_AGENT_INFO_WORKGROUP_MAX_SIZE,
+                             &agent_i->workgroup_max_size);
+
+    // Get max number of work-items of each dimension of a work-group
+    err = hsa_agent_get_info(agent, HSA_AGENT_INFO_WORKGROUP_MAX_DIM,
+                             &agent_i->workgroup_max_dim);
+
+    // Get max number of a grid per dimension
+    err = hsa_agent_get_info(agent, HSA_AGENT_INFO_GRID_MAX_DIM,
+                             &agent_i->grid_max_dim);
+
+    // Get max total number of work-items in a grid
+    err = hsa_agent_get_info(agent, HSA_AGENT_INFO_GRID_MAX_SIZE,
+                             &agent_i->grid_max_size);
+
+    // Get max number of fbarriers per work group
+    err = hsa_agent_get_info(agent, HSA_AGENT_INFO_FBARRIER_MAX_SIZE,
+                             &agent_i->fbarrier_max_size);
+
+    err = hsa_agent_get_info(
+        agent, (hsa_agent_info_t)HSA_AMD_AGENT_INFO_MAX_WAVES_PER_CU,
+        &agent_i->max_waves_per_cu);
+  }
+  return err;
+}
+
+static hsa_status_t getAgentArchInfo(hsa_agent_t agent, void *data) {
+
+  hsa_status_t err;
+  agent_info_t agent_i;
+  std::tuple<std::string, int> *arch_info =
+      reinterpret_cast<std::tuple<std::string, int> *>(data);
+
+  err = AcquireAgentInfo(agent, &agent_i);
+  if (std::string(agent_i.name).rfind("gfx", 0) == 0) {
+    // name length
+    uint32_t name_len;
+    err = hsa_isa_get_info_alt(agent_i.agent_isa, HSA_ISA_INFO_NAME_LENGTH,
+                               &name_len);
+
+    // create buffer
+    isa_info_t isa_i;
+    isa_i.name_str = new char[name_len];
+    if (isa_i.name_str == nullptr) {
+      return HSA_STATUS_ERROR_OUT_OF_RESOURCES;
+    }
+
+    // fill data
+    err = hsa_isa_get_info_alt(agent_i.agent_isa, HSA_ISA_INFO_NAME,
+                               isa_i.name_str);
+
+    // set output tuple
+    std::get<0>(*arch_info) = isa_i.name_str;
+    std::get<1>(*arch_info) = agent_i.wavefront_size;
```

Comment:
Actually I do not see sramecc and xnack functionality in the PR. I see only name and wavefront size.

Reply:
We get this info from the gfx_arch_details.

https://github.com/ROCmSoftwarePlatform/triton/blob/ee0cf024290b512a9201977ab0aee20230708dc1/python/triton/compiler.py#L1738

Reply:
Daniil means that this data is not stored in arch triple.
Features are flags passed to compiler, like `-march=gfx908:sramecc+:xnack-`

For example, I have `gfx1030` on my local machine, which supports following features: `cumode` and `wavefrontsize64` according to [this table](https://www.llvm.org/docs/AMDGPUUsage.html#processors), but rocminfo prints following:
```
$ rocminfo | grep amd
      Name:                    amdgcn-amd-amdhsa--gfx1030
```

Reply:
there does not seem to be a way to get this information (sramecc and xnack settings) using daniil's suggested approach.